_posts/2019-04-03-blog-import-the-style-to-the-typescript-file.md

```diff
@@ -0,0 +1,112 @@
+    }
+}
+```
+However, it has some flaw. The process of the definition file generator need always start a watch by a command like `yarn start`. It also makes our size of the repository become a bigger one although the intellisense is convenient for us.
+
+
+
+### Using the "require" syntax instead of using import.
+
+Using the different style or syntax for import may make our project become a mess, we should keep a consistent form of the import, so maybe there were have a better solution that I not found yet.
+
+
+
+### TypeScript server plugin - [typescript -plugin-css-modules](https://github.com/mrmckeb/typescript-plugin-css-modules)
+This plugin is a TypeScript plugin was inspired by this [create-react-app issue](https://github.com/facebook/create-react-app/issues/5677) and based on css-module-types. It's easy to set up by installing the node module and add plugin to the `tsconfig.json` like the following:
+
+```json
+{
+  "compilerOptions": {
+    "plugins": [{ "name": "typescript-plugin-css-modules" }]
+  }
+}
+```
+
+The match pattern default is `\\.module\\.(sa|sc|c)ss$`. We can custom set other pattern by the property.
+
+```json
+{
+  "compilerOptions": {
+    "plugins": [
+      {
+        "name": "typescript-plugin-css-modules",
+        "options": {
+          "customMatcher": "\\.m\\.css$",
+          "camelCase": "dashes"
+        }
+      }
+    ]
+  }
+}
+```
+After the setting, we also have to change the version of TypeScript. By default, VSCode will use its own version of TypeScript. To make it work correctly with this plugin, there have two options:
+
+1. Add this plugin to "typescript.tsserver.pluginPaths" in settings.
+```json
+{
+  "typescript.tsserver.pluginPaths": ["typescript-plugin-css-modules"]
+}
+```
+But this option is only setable for users setting. It's not convinient for coporation. The next option is more useful for us.
+
+2. Use our workspace's version of TypeScript
+  The workspace's version which will load the plugins from our tsconfig.json` file. We can add the config to the workspace setting.
+
+```json
+{
+  "typescript.tsdk": "./node_modules/typescript/lib"
+}
+```
+
+Besides, we found the issue about import not only style file but also other files like SVG in TypeScript. It will show the error like: "Cannot find module '../../assets/images/logo.svg'.ts(2307)". To resolve the issue that unable to import the SVG file in TypeScript, we can create a definition file named "custom.d.ts" and add the following code and add the file path to the includes field of the `tsconfig.json`:
+```typescript
+declare module "*.svg" {
+  const content: any;
+  export default content;
+}
+```
+
+
+
+### Conclusion
+
+Using the TypeScript server plugin is a good way for the solution that imports the style files which TypeScript can understand. It does not generate the definition file to the project and keeping the repository clean. Furthermore, we can enjoy the benefit of intellisense for CSS selector.
```

Comment:
Using TypeScript server plugin is a good solution for importing style files and assets in TypeScript. It does not generate definition files into the project so that the repository is kept clean. Furthermore, we can also enjoy the benefits of intellisense for CSS selectors.